source/features/conversation-timeline-filter.tsx

```diff
@@ -0,0 +1,174 @@
+import select from 'select-dom';
+import {observe} from 'selector-observer';
+import * as pageDetect from 'github-url-detection';
+import {CheckIcon, GearIcon, XIcon} from '@primer/octicons-react';
+
+import features from '.';
+
+enum FilterSettings {
+	ShowAll = 1,
+	ShowOnlyComments = 2,
+	ShowOnlyUnresolvedComments = 3,
+	ShowOnlyUnresolvedReviews = 4
+}
+
+let currentSettings: FilterSettings = FilterSettings.ShowAll;
+
+const showFilterName = 'rgh-show-filter';
+const menuItemCheckbox = 'rgh-filter-menu-item-checkbox';
+
+// Every element on the timeline that is recognizable by this feature will be marked with tis class.
+const timelineFiltersSelectorId = 'timeline-filters';
+
+const summaries = {
+	[FilterSettings.ShowAll]: '',
+	[FilterSettings.ShowOnlyComments]: 'Show only comments',
+	[FilterSettings.ShowOnlyUnresolvedComments]: 'Show only unresolved reviews',
+	[FilterSettings.ShowOnlyUnresolvedReviews]: 'Show unresolved comments'
+};
+
+function regenerateFilterSummary(): void {
+	select(`#${timelineFiltersSelectorId}`)!.textContent = summaries[currentSettings];
+}
+
+async function saveSettings(filterSettings: FilterSettings, test: string): Promise<any> {
+	currentSettings = filterSettings;
+	regenerateFilterSummary();
+	reapplySettings();
+
+	select.all('.' + menuItemCheckbox)
+		.forEach(element => {
+			element.style.display = 'none';
+		});
+
+	select(test)!.style.display = '';
+}
+
+function reapplySettings(): void {
+	for (const element of select.all('.js-timeline-item')) {
+		processTimelineItem(element);
+	}
+}
+
+// @ts-expect-error TODO: adjust to new select menu
+function restoreSettings(): void {
+	select.all<HTMLInputElement>(`#${showFilterName}`).forEach(element => {
+		element.checked = false;
+	});
+	select<HTMLInputElement>(`input[name=${showFilterName}][value="${currentSettings.valueOf()}"]`)!.checked = true;
+}
+
+function createRadio(title: string, summary: string, filterSettings: FilterSettings, checked: boolean): JSX.Element {
+	return (
+		<label className="select-menu-item d-flex" aria-checked={String(checked)} role="menuitemradio" tabIndex={0} onClick={async () => saveSettings(filterSettings, `#rgh-filter-menu-item-checkbox-${filterSettings}`)}>
+			<CheckIcon id={`rgh-filter-menu-item-checkbox-${filterSettings}`} style={{display: checked ? 'inherit' : 'none'}} className={`${menuItemCheckbox} select-menu-item-icon`} aria-hidden="true"/>
+			<div className="select-menu-item-text">
+				{title}
+				<div className="text-normal description">{summary}</div>
+			</div>
+		</label>
+	);
+}
+
+async function addTimelineItemsFilter(): Promise<void> {
+	select('#partial-users-participants')!.before(
+		<div className="discussion-sidebar-item js-discussion-sidebar-item">
+			<details className="details-reset details-overlay select-menu hx_rsm">
+				<summary className="text-bold discussion-sidebar-heading discussion-sidebar-toggle hx_rsm-trigger" aria-haspopup="menu" data-hotkey="x" role="button">
```

Comment:
Let’s discuss hotkeys in a different issue or PR
```suggestion
				<summary className="text-bold discussion-sidebar-heading discussion-sidebar-toggle hx_rsm-trigger" aria-haspopup="menu" role="button">
```